frontend/lib/resolution/scope-queries.cpp

```diff
@@ -845,19 +842,53 @@ findUseImportTarget(Context* context,
                     UniqueString& foundName) {
   if (auto ident = expr->toIdentifier()) {
     foundName = ident->name();
+
     if (ident->name() == USTR("super")) {
-      return scope->parentScope()->moduleScope();
+
+      // Get the scope of the closest module, or ourself if we are one.
+      auto modScope = scope->moduleScope();
+
+      // Get the next closest module.
+      auto ret = modScope->parentModuleScope();
+
+      // This error works because 'dot' expressions recurse to the left.
+      // If we've hit this case then we're at the leftmost component
+      // in 'super.foo' and failed to find a parent module.
+      // TODO: How difficult would it be to move this to post-parse?
+      if (!ret) CHPL_REPORT(context, SuperFromTopLevelModule, expr);
+
+      return ret;
     } else if (ident->name() == USTR("this")) {
       return scope->moduleScope();
     } else {
       return findScopeViz(context, scope, ident->name(), resolving,
                           expr->id(), useOrImport, /* isFirstPart */ true);
     }
   } else if (auto dot = expr->toDot()) {
-    UniqueString ignoredFoundName;
+    UniqueString ignoreFoundName;
     const Scope* innerScope = findUseImportTarget(context, scope, resolving,
                                                   dot->receiver(), useOrImport,
-                                                  ignoredFoundName);
+                                                  ignoreFoundName);
+    // TODO: 'this.this'?
+    if (dot->field() == USTR("super")) {
+      auto ast = parsing::astToParentAst(context, dot);
+      assert(ast);
+
+      // Ok, descending...
+      if (ast->isDot()) {
```

Comment:
I'm not sure I'm following the logic. Doesn't this restrict us to the case of
```Chapel
import M1.super.bla;
```
Since it expects the parent of the super field access to be a dot expression too? I would think it's okay to just import a `super`.

Reply:
Yep, I think you're right.